Library/Homebrew/env_config.rb

```diff
@@ -132,9 +132,9 @@ module EnvConfig
         boolean:     true,
       },
       HOMEBREW_CURLRC:                           {
-        description: "If set, do not pass `--disable` when invoking `curl`(1), which disables the " \
-                     "use of `curlrc`.",
-        boolean:     true,
+        description: "If set to a path, pass it with `--config` when invoking `curl`(1). " \
+                     "If set but _not_ a valid path, do not pass `--disable`, which disables the " \
+                     "use of `.curlrc`.",
```

Comment:
```suggestion
        description: "If set to a full path (i.e. beginning with `/`, pass it with `--config` when invoking `curl`(1). " \
                     "If set but _not_ a valid path, do not pass `--disable`, which disables the " \
                     "use of `.curlrc`.",
```

note this will fail `brew style`: you'll need to fix up the strings locally. Once that's done, you've run `brew generate-man-completions` and (ideally if your git-fu is sufficient) squashed it down to a single commit and CI is 🟢: this is ready to merge! Sorry for all the back and forth.